bindings/wasm/build/replace_paths.js

```diff
@@ -1,40 +1,53 @@
-const fs = require("fs/promises");
+const fs = require("fs");
 const path = require("path");
 
 /**
- * Replaces aliases defined in `tsconfig.json` files with their corresponding paths.
+ * Replaces aliases defined in the `tsconfig.json` `paths` configuration in `js` and `ts` files.
  * If more than one path is defined. The second path is used. Otherwise the first path.
+ * @param {string} tsconfig - Path to tsconfig that should be processed
+ * @param {string} dist - Folder of files that should be processed
+ * @param {'resolve'=} mode - In "resolve" mode relative paths will be replaced paths relative to the processed file. Note: `basePath` in the tsconfig will not be considered.
  */
-async function main() {
-    if (process.argv[2] === "node") await replace("tsconfig.json", "node");
-    if (process.argv[2] === "web") await replace("tsconfig.web.json", "web");
-}
 
-async function replace(tsconfig, dist) {
+function replace(tsconfig, dist, mode) {
     // Read tsconfig file.
-    let data = JSON.parse(await fs.readFile(path.join(__dirname, `../lib/${tsconfig}`), "utf8"));
+    const tsconfigPath = path.join(__dirname, '..', tsconfig);
+    console.log(`\n using ${tsconfigPath}`);
+    let data = JSON.parse(fs.readFileSync(path.join(__dirname, '..', tsconfig), "utf8"));
     let a = data.compilerOptions.paths;
     let keys = Object.keys(a);
 
     // Get `.js` and `.ts` file names from directory.
-    let files = await fs.readdir(path.join(__dirname, `../${dist}`), { withFileTypes: true });
-    files = files.filter((directoryItem) => directoryItem.isFile()).map((file) => file.name);
+    const distPath = path.join(__dirname, `../${dist}`);
+    console.log(`\n working in ${distPath}`);
+    let files = readdirSync(distPath);
     files = files.filter((fileName) => fileName.endsWith(".ts") || fileName.endsWith(".js"));
 
     // Replace the Alias with the second path if present, otherwise use first path.
     for (let file of files) {
-        let fileData = await fs.readFile(path.join(__dirname, `../${dist}/${file}`), "utf8");
+        console.log(`\n processing ${file}`);
+        let fileData = fs.readFileSync(file, "utf8");
         for (let key of keys) {
             let value = a[key][1] ?? a[key][0];
+            
+            const absoluteIncludePath = path.resolve(path.dirname(tsconfigPath), value);
+            if (mode == "resolve" && fs.existsSync(absoluteIncludePath)) {
+                const absoluteFilePath = path.resolve(path.dirname(file));
+                console.log(`\t calculating path from ${absoluteFilePath} to ${absoluteIncludePath}`);
+                value = path.relative(absoluteFilePath, absoluteIncludePath);
```

Comment:
This doesn't produce the right path in windows since windows uses backslash instead of forward slash in paths. A quick fix could be replacing it.

```suggestion
                value = path.relative(absoluteFilePath, absoluteIncludePath).replaceAll('\\','/');
```